### PR TITLE
RDM-1202 - Case event fields label and hint overrides

### PR DESCRIPTION
**Pull request by gbenadikar**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-1202

### Change description ###
Code changes to support configuring case event fields with label and hint text overrides. Each field used on an event can now have its own label and help text

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: repository/src/main/resources/db/changelog/db.changelog-master.xml [hunk 1]
```diff
@@ -47,6 +47,6 @@
     <include file="db/changelog/db.changelog-rdm-114.xml"/>
     <include file="db/changelog/db.changelog-rdm-2917.xml"/>
     <include file="db/changelog/db.changelog-rdm-2917-cleanup.xml"/>
-
+    <include file="db/changelog/db.changelog-rdm-1202.xml"/>
```
Comment (gbenadikar) [inline repository/src/main/resources/db/changelog/db.changelog-master.xml]: changes to DB are part of separate PR #190

Comment (tzarsmango) [reply repository/src/main/resources/db/changelog/db.changelog-master.xml]: why do they show up here as well?

File: repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/CaseEventField.java [hunk 2]
```diff
@@ -77,4 +79,30 @@ public Integer getShowSummaryContentOption() {
     public void setShowSummaryContentOption(Integer showSummaryContentOption) {
         this.showSummaryContentOption = showSummaryContentOption;
     }
+
+    /**
+     * event case field label
+     **/
+    @ApiModelProperty(required = true, value = "")
```
Comment (tzarsmango) [inline repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/CaseEventField.java]: is this one required or optional?

File: application/src/test/java/uk/gov/hmcts/net/ccd/definition/store/BaseTest.java [hunk 3]
```diff
@@ -56,9 +58,8 @@
 })
 @TestPropertySource(locations = "classpath:test.properties")
 public abstract class BaseTest {
-    public static final String EXCEL_FILE_CCD_DEFINITION = "/CCD_TestDefinition_V37_RDM-2917.xlsx";
+    public static final String EXCEL_FILE_CCD_DEFINITION = "/CCD_TestDefinition_V37_RDM-1202.xlsx";
```
[the remark this answers is not in the document]
Comment (gbenadikar) [reply application/src/test/java/uk/gov/hmcts/net/ccd/definition/store/BaseTest.java]: cool, will do

File: repository/src/main/resources/db/changelog/db.changelog-master.xml (hunk 1 above)
Comment (gbenadikar) [reply repository/src/main/resources/db/changelog/db.changelog-master.xml]: I had to pull the DB branch into this one for successful build (due to changes to Entity), local testing etc.

File: repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/CaseEventField.java (hunk 2 above)
Comment (gbenadikar) [reply repository/src/main/java/uk/gov/hmcts/ccd/definition/store/repository/model/CaseEventField.java]: ah, optional, will change. ta

**Pull request merged**